tree/ntupleutil/v7/src/RNTupleInspector.cxx

```diff
@@ -103,3 +174,130 @@ float ROOT::Experimental::RNTupleInspector::GetCompressionFactor()
 {
    return (float)fUncompressedSize / (float)fCompressedSize;
 }
+
+int ROOT::Experimental::RNTupleInspector::GetFieldTypeFrequency(std::string className)
+{
+   if (fFieldTypeFrequencies.empty()) {
+      auto descriptorGuard = fPageSource->GetSharedDescriptorGuard();
+
+      for (const auto &fieldDescriptor : descriptorGuard->GetTopLevelFields()) {
+         fFieldTypeFrequencies[fieldDescriptor.GetTypeName()]++;
+      }
+   }
+
+   return fFieldTypeFrequencies[className];
+}
+
+int ROOT::Experimental::RNTupleInspector::GetColumnTypeFrequency(ROOT::Experimental::EColumnType colType)
+{
+   if (fColumnInfo.empty()) {
+      CollectColumnData();
+   }
+
+   int typeFrequency = 0;
+
+   for (auto const &colInfo : fColumnInfo) {
+      if (colInfo.fType == colType) {
+         ++typeFrequency;
+      }
+   }
+
+   return typeFrequency;
+}
+
+std::uint64_t ROOT::Experimental::RNTupleInspector::GetOnDiskColumnSize(ROOT::Experimental::DescriptorId_t logicalId)
+{
+   if (fColumnInfo.empty()) {
+      CollectColumnData();
+   }
+
+   auto descriptorGuard = fPageSource->GetSharedDescriptorGuard();
+
+   if (logicalId > descriptorGuard->GetNLogicalColumns()) {
+      std::cerr << "no column with id " << logicalId << " present" << std::endl;
```

Comment:
I guess we should be throwing an exception instead in this case (in principle, the same applies also to all the prints below).

Reply:
I think a simple exception is fine.  The `R__FAIL` / `RResult` failure handling is used for deep call stacks where you are interested in a stack track on an exception and where you would like to be able to use return-value based error handling in addition to exception handling.